source/geometry_model/initial_topography_model/prm_polygon.cc

```diff
@@ -0,0 +1,157 @@
+/*
+  Copyright (C) 2011 - 2016 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/geometry_model/initial_topography_model/prm_polygon.h>
+#include <aspect/utilities.h>
+#include <boost/lexical_cast.hpp>
+
+namespace aspect
+{
+  namespace InitialTopographyModel
+  {
+    template <int dim>
+    double
+    PrmPolygon<dim>::
+    value (const Point<dim-1> &p) const
+    {
+      double value = 0;
+      Point<2> p1;
+
+      if (dim == 2)
+        p1 = Point<2>(p[0],0);
+      else
+        p1 = Point<2>(p[0],p[1]);
+
+      for (unsigned int i = 0; i < point_lists.size(); i++)
+        {
+          if (aspect::Utilities::polygon_contains_point<dim>(point_lists[i],p1))
+            {
+              value = topography_values[i];
+            }
+        }
+      return value;
+    }
+
+
+
+    template <int dim>
+    void
+    PrmPolygon<dim>::
+    declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Geometry model");
+      {
+        prm.enter_subsection("Initial topography");
+        {
+          prm.enter_subsection("Prm polygon");
+          {
+            prm.declare_entry("Topography parameters",
+                              "",
+                              Patterns::Anything(),
+                              "Set the topography height and the polygon which should be set to that height. "
+                              "The formate is : \"The topography height > The point list describing a polygon & "
+                              "The next topography height > the next point list describing a polygon.\" "
+                              "The format for the point list describing the polygon is \"x1,y1;x2,y2\". "
+                              "For example for two triangular areas of 100 and -100 meters high set: "
+                              "'100 > 0,0;5,5;0,10 & -100 > 10,10;10,15;20,15'. "
+                              "Units of the height are always in meters. The units of the coorindates are "
+                              "dependent on the geometry model.");
```

Comment:
Can you be more specific here? E.g., for the 'box' model, it is interpreted in meters; for spherical models, point coordinates are interpreted as latitude/longitude? Or do I entirely misinterpret what you're trying to say.